### PR TITLE
Add linux job types.

### PR DESCRIPTION
**Pull request by inferno-chromium**

### CODE REVIEW

File: tool/clusterfuzz/binary_providers.py [hunk 1]
```diff
@@ -334,9 +334,7 @@ def __init__(self, testcase, binary_definition, current, goma_dir,
     self.name = 'V8'
 
   def pre_build_steps(self):
-    if not self.current:
-      common.execute('gclient', 'runhooks', self.source_directory)
-    common.execute('gypfiles/gyp_v8', '', self.source_directory)
```
Comment (tanin47) [inline tool/clusterfuzz/binary_providers.py]: I didn't know we didn't need this. Thank you.

File: tool/clusterfuzz/binary_providers.py [hunk 2]
```diff
@@ -359,8 +357,8 @@ def __init__(self, testcase, binary_definition, current, goma_dir,
     self.name = 'chromium'
 
   def pre_build_steps(self):
+    common.execute('gclient', 'runhooks', self.source_directory)
```
Comment (tanin47) [inline tool/clusterfuzz/binary_providers.py]: What is the reason for moving this out?

`--current` means that the user ensures the repo is in a good state and ready to be built.

Comment (inferno-chromium) [inline tool/clusterfuzz/binary_providers.py]: If the repo is not in good state, it fails with a bad silent error. Running runhooks just brings the dependencies, there is nothing custom in there, like custom clang.

Comment (tanin47) [inline tool/clusterfuzz/binary_providers.py]: Ok, that's fair. The deal breaker is `gclient sync` anyway, not `runhooks`.

**Pull request merged**